jetty-core/jetty-http2/jetty-http2-server/src/main/java/org/eclipse/jetty/http2/server/internal/HttpStreamOverHTTP2.java

```diff
@@ -221,7 +221,10 @@ else if (!_demand)
         {
             Runnable task = _httpChannel.onContentAvailable();
             if (task != null)
-                _connection.offerTask(task, false);
+            {
+                // We must dispatch, so an application thread does not become a producer and then consume another request.
```

Comment:
Just compacting the comment.

```suggestion
                // We must dispatch, so an application thread does not 
                // become a producer and then consume another request.
```